pkg/signup/signup_service.go

```diff
@@ -238,6 +242,45 @@ func (s *ServiceImpl) GetSignup(userID string) (*Signup, error) {
 	return signupResponse, nil
 }
 
+// PostVerification returns userSignup resource from the host cluster.
+// Returns the usersignup, an http code and an error if present.
+func (s *ServiceImpl) PostVerification(dailyLimit int, responseBody map[string]string, userID, code string) (*crtapi.UserSignup, int, error) {
+	// Retrieve UserSignup resource from the host cluster
+	userSignup, err := s.UserSignups.Get(userID)
+	if err != nil {
+		return nil, http.StatusNotFound, err
+	}
+
+	// get the annotation counter
+	annotationCounter := userSignup.Annotations[crtapi.UserSignupVerificationCounterAnnotationKey]
+	counter := 0
+	if annotationCounter != "" {
+		counter, err = strconv.Atoi(annotationCounter)
+		if err != nil {
+			return nil, http.StatusInternalServerError, err
+		}
+	}
+
+	// check if counter has exceeded the limit of daily limit - if at limit error out
+	if counter > dailyLimit {
+		return nil, http.StatusForbidden, errors2.New("daily limit has been exceeded")
```

Comment:
Let's log the current counter and the limit.

Reply:
fixed here: https://github.com/codeready-toolchain/registration-service/blob/7908cfab4b879d2a292e2d8dafb882a6bbe3b967/pkg/verification/verification_service.go#L103